lrauv_ignition_plugins/include/lrauv_ignition_plugins/comms/CommsClient.hh

```diff
@@ -67,15 +74,58 @@ public: CommsClient(uint32_t _address,
 /// \param[in] _msg - Comms message to be sent
 public: void SendPacket(const CommsMsg& _msg)
 {
-  this->transmitter.Publish(_msg);
+  // Populate a Dataframe msg based on the
+  // CommsMsg content.
+  gz::msgs::Dataframe message;
+  message.set_src_address(
+      std::to_string(_msg.from()));
+  message.set_dst_address(
+      std::to_string(_msg.to()));
+  message.set_data(_msg.data());
+
+  // Set message type
+  auto *frame = message.mutable_header()->add_data();
+  frame->set_key("msg_type");
+  if (_msg.type() == LRAUVAcousticMessageType)
+    frame->add_value("LRAUVAcousticMessageType");
+  else if (_msg.type() == RangeRequest)
+    frame->add_value("RangeRequest");
+  else if (_msg.type() == RangeResponse)
+    frame->add_value("RangeResponse");
```

Comment:
@adityapande-1995 nit: why not use an `std::map` or some other form of associative container. Also, should we throw if the type is unknown? Same below (in reverse).

Reply:
Fixed here : https://github.com/osrf/lrauv/pull/243/commits/75fe73412f8c588191bfe05640115556d6abb973